src/tabs/TabModel.tsx

```diff
@@ -53,8 +53,10 @@ export abstract class TabModel<StateType> {
     this.updateView(state);
   }
 
-  public requestResource = (data: ResourceRequest) =>  {
-    this.sendData({messageType: "RESOURCE_REQUEST", data});
+  public requestResource = (data: ResourceRequest) => {
+    if (this.sendData instanceof Function) {
```

Comment:
I don't really understand why this check is necessary, can you explain the reason which lead you to introduce it? I suspect it's because something it's not always initialized correctly - which are the conditions that make this happen? It may indicate some flaw in the code which handles the various resource updates, requests, etc (my code and my fault, probably)

Reply:
I was experimenting with refreshing the frontend in this PR, and it looks like I forgot to remove this line. The reason why I needed to add it was because the frontend was trying to refresh before initializing everything, causing errors. I will change it back.

Reply:
Ah, okay